### DISCUSSION TITLE
StreamFX 0.12.0 Alpha 77

### DISCUSSION THREAD

**Discussion by Xaymar** [Xaymar/obs-stream-effects#882] [Announcements]
# ⚠ Preview Release not for Production! ⚠
This release is meant for testing in isolated conditions and early adopters only, and should never be used in production. If you use one of these releases, all promises of support, warranty, functionality, and similar are immediately void and if things break you are on your own. [The latest production ready release can be found here](https://github.com/Xaymar/obs-StreamFX/releases/latest).

# Please read the [Installation Guide](https://github.com/Xaymar/obs-StreamFX/wiki/Installation) and [System Requirements](https://github.com/Xaymar/obs-StreamFX/wiki/System-Requirements)!
The installation of StreamFX requires that you follow certain instructions, as well as fulfill a minimum hardware and software requirement. Please ensure that you have read and verified both before asking for help, as many issues can be fixed by reading both of them in their entirety.

# Support StreamFX on Github Sponsors or Patreon!
[<img align="left" alt="GitHub Sponsors" src="https://user-images.githubusercontent.com/437395/106463119-48412400-6497-11eb-9934-baed8e0bdd73.png" height="70px"/>](https://github.com/sponsors/xaymar) [<img align="right" alt="Patreon" src="https://user-images.githubusercontent.com/437395/106462708-bd602980-6496-11eb-8f35-038577cf8fd7.jpg" height="70px"/>](https://patreon.com/xaymar) The StreamFX project is only possible through your support, either as a [translator](https://github.com/Xaymar/obs-StreamFX/issues/36) or as a Supporter on either [Patreon](https://patreon.com/xaymar) or [Github](https://github.com/sponsors/xaymar). Your support helps keep StreamFX free from Ads, even if it's just translating a few lines per month, or 1$ per month. If continuous support isn't your thing, maybe a one-time donation via [PayPal](https://paypal.me/xaymar) is?

# Overview of Changes
## New Release process: Alpha -> Beta -> Candidate -> Release!
The old two stage system didn't work out quite as I'd have hoped: It didn't communicate the dangers of Testing well, and did not differentiate between different stability levels. To solve this I've adopted a new release process now available in StreamFX, which follows the usual development cycle of Software itself: Alpha -> Beta -> Candidate -> Release. The further right you go in that graph, the more stable software is supposed to be, at the cost of new features and changes. The further left you go, the more up-to-date things will be, but you sacrifice stability and safety.

## Avid DNxHR Encoder (via FFmpeg) by @IceStormNG
A new contender for the visually lossless intermediate codec arrives: Avid DNxHR! Thanks to work done by @IceStormNG, it is now possible to use this encoder without configuration trickery and directly configure it with a proper user interface. As this encoder is based on the real codec specification, it may perform better than the Apple ProRes encoder while being way more compatible with video editing software.

## ARM64 Support on MacOS
As OBS Studio 28.0 and beyond support ARM64 and x86-64 now, StreamFX now also supports this architecture natively. Note that you may need at least MacOS 11.0 for the ARM64 binaries to load correctly.

## [Detailed Changelog](https://github.com/Xaymar/obs-StreamFX/compare/0.11.0...0.12.0a77)
<details><summary>0.12.0a77</summary>

> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/817
> * Improve GitHub integration by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/841
> * project: Fix incorrect tabs in YAML by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/842
> * Improve metadata generation by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/837
> * cmake: Add support for Qt6 by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/866
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/832
> * Further improve CI and Github integration by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/843
> * ci: Remove Code Signing with self-signed certificate by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/873
> * Fix compilation errors with Debug and/or Profiling by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/876
> * Deduplicate CI build pipeline by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/874
> * Remove Auto-Dependency System by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/870
> * Working on source code [Refactor] by @GermanAizek in https://github.com/Xaymar/obs-StreamFX/pull/836
> * obs/gs/effect: Revert #836 by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/880
> * Update libOBS to 28.0.0-rc1 by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/878
</details>

<details><summary>0.12.0a45</summary>

> * Improve CMake build script(s) by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/813
> * ci: Add GCC back to the build by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/820
> * Add ClangCL, and update Windows SDK requirements by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/821
> * ci: Allow failures for experimental CI tests by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/824
> * ci: Use specific LLVM/Clang versions by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/825
> * project: Add support for multi stage release cycles by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/823
</details>

<details><summary>0.12.0a1</summary>

> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/778
> * ffmpeg-encoder/dnxhr: Added Avid DNxHR Encoder based on FFmpeg by @IceStormNG in https://github.com/Xaymar/obs-StreamFX/pull/784
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/786
> * Refactor C++ wrappers for libOBS by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/806
</details>

# System Requirements
<table style="table-layout: fixed">
  <tr>
    <th width="5%"><img width=10000000px height=1 /></th>
    <th width="5%"></th>
    <th center width="30%"><img src="https://streamfx.xaymar.com/assets/platforms/platform-windows.svg" alt="Windows"><br>Windows</th>
    <th center width="30%"><img src="https://streamfx.xaymar.com/assets/platforms/platform-linux.svg" alt="Linux"><br>Ubuntu, Debian, ...</th>
    <th center width="30%"><img src="https://streamfx.xaymar.com/assets/platforms/platform-macos.svg" alt="MacOS"><br>MacOS</th>
  </tr>
  <tr>
    <th rowspan=2>OS</th>
    <th>Minimum</th>
    <td>Windows 10 21H2 (or newer)</td>
    <td>Ubuntu 20.04 (or newer), (or equivalent Distro)</td>
    <td>MacOS 10.15 (or newer)</td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td>Windows 10 21H2 (or newer), Windows 11 (or newer)</td>
    <td>Ubuntu 22.04 (or newer), <i>or equivalent Distro</i></td>
    <td>MacOS 12 (or newer)</td>
  </tr>
  <tr>
    <th rowspan=4>CPU</th>
    <th rowspan=2>Minimum</th>
    <td colspan=3>
      <i>Any x86-64-v2 compatible:</i> <a href="https://en.wikipedia.org/wiki/Excavator_(microarchitecture)">AMD Excavator Series</a>, <a href="https://en.wikipedia.org/wiki/Haswell_(microarchitecture)">Intel Haswell Series</a>
    </td>
  </tr>
  <tr>
    <td colspan=2 style="display: none;"></td>
    <td>
      <i>Any ARM64-v8.5-A compatible:</i> <a href="https://en.wikipedia.org/wiki/Apple_M1">Apple M1 Series</a>
    </td>
  </tr>
  <tr>
    <th rowspan=2>Recommended</th>
    <td colspan=3>
      <i>Any x86-64-v2 compatible:</i> <a href="https://en.wikipedia.org/wiki/Zen_2">AMD Zen 2 Series</a>, <a href="https://en.wikipedia.org/wiki/Comet_Lake_(microprocessor)">Intel Comet Lake Series</a>
    </td>
  </tr>
  <tr>
    <td colspan=2 style="display: none;"></td>
    <td>
      <i>Any ARM64-v8.5-A compatible:</i> <a href="https://en.wikipedia.org/wiki/Apple_M1">Apple M1 Series</a>
    </td>
  </tr>
  <tr>
    <th rowspan=2>Memory (RAM)</th>
    <th>Minimum</th>
    <td colspan=3>4 GiB (free)</td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td>8 GiB (free)</td>
    <td colspan=2>6 GiB (free)</td>
  </tr>
  <tr>
    <th rowspan=2>Video (GPU)</th>
    <th>Minimum</th>
    <td colspan=3>
      DirectX 11 / OpenGL 3 capable, Shader Model 4, 4 GiB VRAM (total)
    </td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td colspan=3>
      DirectX 12 / OpenGL 4 capable, Shader Model 5, 6 GiB VRAM (total)
    </td>
  </tr>
  <tr>
    <th rowspan=2>Disk Space</th>
    <th>Minimum</th>
    <td colspan=3>~50 MiB (free)</td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td colspan=3>~50 MiB (free)</td>
  </tr>
  <tr>
    <th rowspan=2>OBS Version</th>
    <th>Minimum</th>
    <td colspan=3>28.0.0-rc1</td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td colspan=3>28.0.0-rc1 or <a href="https://obsproject.com/" target="_blank">Latest</a></td>
  </tr>
  <tr>
    <th colspan=2 center>Other</th>
    <td>
      <a href="https://aka.ms/vs/17/release/vc_redist.x64.exe">Microsoft Visual C/C++ Redistributable 2017-2022</a>
    </td>
    <td>
      Qt for obs-studio<br>
      FFmpeg for obs-studio<br>
    </td>
    <td>
    </td>
  </tr>
</table>

<hr /><em>This discussion was created from the release <a href='https://github.com/Xaymar/obs-StreamFX/releases/tag/0.12.0a77'>StreamFX 0.12.0 Alpha 77</a>.</em>

**Comment by LuminairPrime** [comment]
Congrats for the update to the new Qt and support for OBS 28 RC!

**Comment by syropian** [comment]
Great to see native M1 support! Unfortunately the plugin does not seem to show up in the latest OBS 28 RC (RC2 at time of writing). There were no installation errors, but no sign of it anywhere in OBS either. I realize this is an alpha, so maybe this is known/expected 😄

**Reply by Xaymar** [reply]
https://github.com/Xaymar/obs-StreamFX/issues/883
